pyquil/_parser/parser.py

```diff
@@ -451,7 +451,7 @@ def classical_comparison(self, op, target, left, right):
     def waveform(self, name, *params):
         param_dict = {k: v for (k, v) in params}
         if param_dict:
-            return _wf_from_dict(name, param_dict)
+            return TemplateWaveform(name, **param_dict)
```

Comment:
Had to cleanup the `_wf_from_dict` import to keep the tests from crashing, and it was easy enough to patch in this fix, even though we're probably going to delete this parser.